docs/docs/0.6.0/documentation/networking.md

```diff
@@ -87,4 +87,101 @@ In order for connectivity to work among the `kuma-dp` instances, Kuma also assum
+:::
+
+### Configuration
+
+Kuma DNS can be configured by the configuration file, or by the respective environment variables as follows:
+
+```yaml
+# DNS Server configuration
+dnsServer:
+  # The domain that the server will resolve the services for
+  domain: "mesh" # ENV: KUMA_DNS_SERVER_DOMAIN
+  # Port on which the server is exposed
+  port: 5653 # ENV: KUMA_DNS_SERVER_PORT
+  # The CIDR range used to allocate
+  CIDR: "240.0.0.0/4" # ENV: KUMA_DNS_SERVER_CIDR
+```
+
+The `domain` field can change the default `.mesh` DNS zone that Kuma DNS will resolve for. If this is changed, please check the output of `kumactl install dns` and change the zone accordingly, so that your Kube DNS or Core DNS server will redirect all the relevant DNS requests.
+
+The `port` can set the port on which the Kuma DNS is accepting requests. Changing this value on Kubernetes shall be reflected in the respective port setting in the `kuma-control-plane` service. 
+
+The `CIDR` field sets the IP range of virtual IPs. The default `240.0.0.0/4` is reserved for future use IPv4 range and is guaranteed to be non-routable. We strongly recommend to not change this, unless it is really needed.
+
+### Operation 
+
+The basic operation of Kuma DNS includes a couple of main components: DNS server, VIPs allocator, cross-replica persistence.
+
+The DNS server listens on port `5653` and reponds for type `A` DNS requests and answers with `A` record, e.g. `<service>.mesh.	60	IN	A	240.0.0.100`. The default TTL is set to 60 seconds, to ensure client will synchronize with Kuma DNS and account for any changes happening meanwhile.
+
+Kuma DNS allocates the virtual IPs from the configured CIDR, by constantly scanning the services available in all Kuma meshes. When a service is removed its VIP is freed too and Kuma DNS will not respond for it with `A` DNS record.
+
+Kuma DNS is optimized to work in a multi-replica setup, where a number of Kuma control planes are running in parallel. It leverages the internal leader election so that VIP allocation happens only on the leader, and uses an internal storage resource to ensure all other replicas are in sync with the available service/VIP mappings allocated by the leader.
+
+### Usage
+
+Consuming a service handled by Kuma DNS from inside a Kubernetes container looks like this:
+```bash
+<kuma-enabled-pod>$ curl http://echo-server_echo-example_svc_1010.mesh:80
+<kuma-enabled-pod>$ curl http://echo-server_echo-example_svc_1010.mesh
+```
+
+Since the default VIP created listenere will default to port `80`, it can be omitted when using a standard HTTP client.
+
+Effectively the result of the Kuma DNS operation is creating a virtual IP listener by patching the dataplane resource before it gets translated to Envoy xDS configuration. However, by inspecting `curl localhost:9901/config_dump`, we can see sections similar to this one:
```

Comment:
`patching dataplane resource` seems like again - implementation detail, especially that we don't really store patched dataplane. How about something like:

Kuma allocates IP for every Service withing a mesh. Then, it creates outbound virtual listener for every VIP.